src/odemis/driver/_mccdaq.py

```diff
@@ -0,0 +1,700 @@
+    BP_1_00V = 0x70  # Differential +/- 1.00 V
+
+    # Option values for AInScan
+    AIN_EXECUTION = 0x1  # 1 = single execution, 0 = continuous execution
+    AIN_BURST_MODE = 0x2  # 1 = Burst Mode
+    AIN_TRANSFER = 0x4  # 1 = Block Transfer Mode
+    AIN_TRIGGER = 0x8  # 1 = Use External Trigger
+
+    # Commands and Codes for USB 1208-LS HID reports
+    # Digital I/O Commands
+    DCONFIG = 0x0D  # Configure digital port
+    DIN = 0x00  # Read digital port
+    DOUT = 0x01  # Write digital port
+    DBIT_IN = 0x02  # Read digital port bit
+    DBIT_OUT = 0x03  # Write digital port bit
+
+    # Analog Input Commands
+    AIN = 0x06  # Read analog input channel
+    AIN_SCAN = 0x0E  # Scan analog channels
+    AIN_STOP = 0x10  # Stop scan
+    ALOAD_QUEUE = 0x07  # Load the channel/gain queue
+
+    # Analog Output Commands
+    AOUT = 0x08  # Write analog output channel
+
+    # Counter Commands
+    CINIT = 0x05  # Initialize counter
+    CIN = 0x04  # Read Counter
+
+    # Memory Commands
+    MEM_READ = 0x09  # Read Memory
+    MEM_WRITE = 0x0A  # Write Memory
+
+    # Miscellaneous Commands
+    BLINK_LED = 0x0B  # Causes LED to blink
+    RESET = 0x11  # Reset USB interface
+    SET_TRIGGER = 0x14  # Configure external trigger
+    SET_ID = 0x0C  # Set the user ID
+    GET_ID = 0x0F  # Get the user ID
+
+    scanIdx = 0  # scan index
+    productID = 0  # product ID
+
+    def __init__(self):
+        super().__init__()
+        try:
+            self.h = hid.device()
+        except:
+            print('Error creating hid device')
+
+    #################################
+    #     Digital I/O  Commands     #
+    #################################
+
+    def DConfig(self, port_number, bit_mask):
+        '''
+        This command sets the direction of the digital bits for a port
+
+         port_number:    AUXPORT    =   0x10
+                         Port A     =   0x01
+                         Port B     =   0x04
+
+         bit_mask   bit value:  0 = output,  1 = input
+        Note: Bug in AUXPORT, take the one's complement
+        '''
+
+        if self.productID == 0x0075 and port_number == self.DIO_AUXPORT:
+            bit_mask = ((bit_mask ^ 0xff) & 0xff)
+        self.h.write([self.DCONFIG, port_number, bit_mask, 0, 0, 0, 0, 0])
+
+    def DIn(self, port_number):
+        '''
+        This command reads the current state of the DIO port.  If the digital port
+        is bitwise configurable, it may be any combination of inputs and outputs.
+        The return value will be the value seen at the port pins.
+
+         port_number:    AUXPORT    =   0x10
+                         Port A     =   0x01
+                         Port B     =   0x04
+        Bug: need leading 0 in command string.
+        '''
+
+        self.h.write([0x0, self.DIN, port_number, 0, 0, 0, 0, 0, 0])
+        try:
+            value = self.h.read(8, 500)
+        except:
+            print('DIn: error in reading.')
```

Comment:
This type of except + print is really not helpful. The print will be lost in the darkness of backend stdout, and immediately after `value[0]` will fail as `value` is not defined, and hide the actual error. I'd rather just drop the whole try/except (for all the commands).

Reply:
I'd rather not tinker too much on GH code, but I attempted to remove a few.